### PR TITLE
Added support for custom type on array fields Fixes #14 and slices of…

### LINKED ISSUE #12: Support Slice of Pointers
Need to add implementation and test case for []*Address.

### LINKED ISSUE #14: AddSimpleArrayField ignores ts_type
Need to add customTSType logic similar to AddSimpleField

### PR DESCRIPTION
**Pull request by sampaioletti**
… pointers Fixes #12
Here is an initial PR for #12 and #14 please have a look. I added the scenarios to the main go test file. But didn't do anything with browser test. If you give me some direction on how you might like that integrated I can do that as well. Had to add another hook checking if typescriptChunk is empty, not sure why i'm having to do that if you don't..any ideas? (Using go1.10.1 windows/amd64)

### CODE REVIEW

**Comment by sampaioletti** [comment]
Just noticed I left some log lines in there I was using for debug.  I'll submit a commit to fix those when I get a chance.

File: typescriptify/typescriptify.go [hunk 1]
```diff
@@ -295,13 +300,20 @@ type typeScriptClassBuilder struct {
...
-			t.fields += fmt.Sprintf("%s%s: %s[];\n", t.indent, fieldName, typeScriptType)
-			t.createFromMethodBody += fmt.Sprintf("%s%sresult.%s = source[\"%s\"];\n", t.indent, t.indent, fieldName, fieldName)
-			return nil
-		}
+func (t *typeScriptClassBuilder) AddSimpleArrayField(fieldName string, field reflect.StructField) error {
+	fieldType, kind := field.Type.Elem().Name(), field.Type.Elem().Kind()
+
+	customTSType := field.Tag.Get(tsType)
+
+	typeScriptType := t.types[kind] + "[]"
```
[the remark this answers is not in the document]
Comment (sampaioletti) [reply typescriptify/typescriptify.go]: Proposed Fix in 6f5c83d, Should we be throwing a different error if len(fieldName) >0 ==false? In what situation could this occur....Just a thought

**Comment by sampaioletti** [comment]
Sorry I didn't make the modifications in your array-ptr branch I didn't see that one before I started this PR. I just noticed it in your comment on #17. Next time i'll look around better (:

**Comment by sampaioletti** [comment]
Sorry for the bad commit, fat fingered it somewhere along the line.

**Comment by tkrajina** [comment]
Don't worry, fat-fingered bad commits happen to me all the time :)

Merged not, thanks!

**Pull request merged**